src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopies = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      stateCopy = addProperties(stateCopy, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      stateCopy = removeProperties(stateCopy, action.keysToRemove);
+    } else if (action.type === 'clear') {
+      stateCopy = clear(stateCopy);
+    }
```

Comment:
A default case should be added in the switch statement to handle any unexpected action types, which can help with error handling and make the code more robust.